src/ice/agent.rs

```diff
@@ -1454,6 +1460,15 @@ impl IceAgent {
             }
         }
     }
+
+    pub(crate) fn remote_credentials_changed(&self, new_creds: Option<&IceCreds>) -> bool {
+        // Before we have set remote credentials we don't consider anything a change.
+        let Some(current) = &self.remote_credentials else {
+            return false
+        };
+
+        Some(current) != new_creds
+    }
```

Comment:
Make a `remote_credentials(&self) -> Option<&IceCreds>` and make the comparison on the sdp.rs side.

Reply:
I did that initially, but felt it was nicer to have this method since it does a bit more than 

```rust
ice.remote_credentials() != credentials
```

In particular there's this special case about credentials not being considered changed if they have never been set

Reply:
I think I prefer that logic not being inside here. Unless we unconditionally set it and detect changes in `set_remote_credentials`

Reply:
addressed in e794138